### DISCUSSION TITLE
How to do a global search of a selection?

### DISCUSSION THREAD

**Discussion by ksubramz** [helix-editor/helix#1933] [Q&A]
Ideally, I want to be able to assign a key to search the entire project for the word under cursor. 

Right now, it seems I can enter SPC + / and then type the text to do a global search.

**Comment by Zeddicus414** [comment]
I've been trying to answer the same question. I think it's not possible right now.

**Comment by rcorre** [comment]
Currently you can press `SPC+/`, then `ctrl+s` to insert the selection, or press `*` to set the search register to the selection, then press `SPC+/` and it will be pre-filled with the search.

I tried `C-g = ["search_selection", "global_search"]`, which opens the global search prompt and inserts the selection, but you still have to press `enter`. There also seems to be a visual bug: if I have the text "three" selected, it sets the search to `three 'three'`, but pressing `enter` still seems to search for "three".

**Reply by Valadares-A** [reply]
Yeah, i did the same key binding and notice the same "visual bug"

**Reply by green-lad** [reply]
I've put it into a macro keybinding mapped to `ctrl+/` (using wezterm I had to set `enable_kitty_keyboard=true` in its config, see [this](https://github.com/helix-editor/helix/issues/1369) and [that](https://github.com/wezterm/wezterm/discussions/5472)):
`"C-/" = '@"/y /<ret>'`

This does not escape the search, so I mapped `search_selection` to `alt+/`: `"A-/" = "search_selection"`
And used that which works great: `"C-/" = '@<A-/> /<ret>'`

For searching sth from the system clipboard I use (`Alt-d` deletes stuff without copying it into register `"`) : `"C-?" = '@"+p<A-/><A-d> /<ret>'`
Be cautious with newlines since they are not shown in the global search picker and do not get escaped right.
